pkg/planner/core/indexmerge_path.go

```diff
@@ -1428,18 +1473,26 @@ func PrepareCols4MVIndex(
 // collectFilters4MVIndex splits these filters into 2 parts where accessFilters can be used to access this index directly.
 // For idx(x, cast(a as array), z), `x=1 and (2 member of a) and z=1 and x+z>0` is split to:
 // accessFilters: `x=1 and (2 member of a) and z=1`, remaining: `x+z>0`.
-func collectFilters4MVIndex(sctx context.PlanContext, filters []expression.Expression, idxCols []*expression.Column) (accessFilters, remainingFilters []expression.Expression) {
+func collectFilters4MVIndex(
+	sctx context.PlanContext,
+	filters []expression.Expression,
+	idxCols []*expression.Column,
+) (accessFilters, remainingFilters []expression.Expression, accessTp int) {
 	usedAsAccess := make([]bool, len(filters))
 	for _, col := range idxCols {
 		found := false
 		for i, f := range filters {
 			if usedAsAccess[i] {
 				continue
 			}
-			if checkFilter4MVIndexColumn(sctx, f, col) {
+			if ok, tp := checkFilter4MVIndexColumn(sctx, f, col); ok {
 				accessFilters = append(accessFilters, f)
 				usedAsAccess[i] = true
 				found = true
+				// access filter type on mv col overrides normal col for the return value of this function
+				if accessTp == unspecifiedFilterTp || accessTp == eqOnNonMVColTp {
```

Comment:
why use accessTp to do the comparson？
it isn't be assigned with any value before？